src/java.base/share/classes/java/text/ListFormat.java

```diff
@@ -0,0 +1,571 @@
+
+    private static final Pattern PARSE_START = Pattern.compile("(.*?)\\{0}(.*?)\\{1}");
+    private static final Pattern PARSE_MIDDLE = Pattern.compile("\\{0}(.*?)\\{1}");
+    private static final Pattern PARSE_END = Pattern.compile("\\{0}(.*?)\\{1}(.*?)");
+    private transient Pattern startPattern;
+    private transient String middleBetween;
+    private transient Pattern endPattern;
+
+    private ListFormat(Locale l, String[] patterns) {
+        locale = l;
+        this.patterns = patterns;
+        init();
+    }
+
+    private void init() {
+        // get pattern strings
+        var m = PARSE_START.matcher(patterns[START]);
+        String startBefore;
+        String startBetween;
+        if (m.matches()) {
+            startBefore = m.group(1);
+            startBetween = m.group(2);
+        } else {
+            throw new IllegalArgumentException("start pattern is incorrect: " + patterns[START]);
+        }
+        m = PARSE_MIDDLE.matcher(patterns[MIDDLE]);
+        if (m.matches()) {
+            middleBetween = m.group(1);
+        } else {
+            throw new IllegalArgumentException("middle pattern is incorrect: " + patterns[MIDDLE]);
+        }
+        m = PARSE_END.matcher(patterns[END]);
+        String endBetween;
+        String endAfter;
+        if (m.matches()) {
+            endBetween = m.group(1);
+            endAfter = m.group(2);
+        } else {
+            throw new IllegalArgumentException("end pattern is incorrect: " + patterns[END]);
+        }
+
+        // generate two/three patterns, if empty
+        if (patterns[TWO] == null || patterns[TWO].isEmpty()) {
+            patterns[TWO] = startBefore + "{0}" + endBetween + "{1}" + endAfter;
+        }
+        if (patterns[THREE] == null || patterns[THREE].isEmpty()) {
+            patterns[THREE] = startBefore + "{0}" + startBetween + "{1}" + endBetween + "{2}" + endAfter;
+        }
+        startPattern = Pattern.compile(startBefore + "(.+?)" + startBetween);
+        endPattern = Pattern.compile(endBetween + "(.+?)" + endAfter);
+    }
+
+    /**
+     * {@return the available locales that support list formatting}
+     */
+    public static Locale[] getAvailableLocales() {
+        // Same as a typical format class
+        return DateFormat.getAvailableLocales();
+    }
+
+    /**
+     * {@return the list format object for the default
+     * {@link Locale.Category#FORMAT FORMAT Locale}, {@code STANDARD} type,
+     * and {@code FULL} style}
+     */
+    public static ListFormat getInstance() {
+        return getInstance(Locale.getDefault(Locale.Category.FORMAT), Type.STANDARD, Style.FULL);
+    }
+
+    /**
+     * {@return the list format object for the specified {@code Locale}, {@code Type},
+     * and {@code Style}}
+     * @param locale Locale to be used, not null
+     * @param type type of the list format. One of STANDARD/OR/UNIT, not null
+     * @param style style of the list format. One of FULL/SHORT/NARROW, not null
```

Comment:
It would be ease navigation in the javadoc if there was a link to the ListFormat.Type and ListFormat.Style classes, perhaps Locale also.
Use the normal prose form STANDARD, OR, or UNIT, instead of the "/" shorthand. Here and throughout the javadoc.  Ditto FULL, SHORT, or NARROW.
Also use Locale, Type, and Style instead of the shorthand.